### PR TITLE
Fix: 7451

### LINKED ISSUE #7451: Synergy 1 is out of bound
### Discussion

- [X] Yes, and I still want to report a bug

### What type of Synergy user are you?

Community user

### Support ticket number (for customers)

_No response_

### Product name

Synergy 1

### Synergy version number

1.15.0

### Operating systems (OS)

- [X] Windows
- [ ] macOS
- [ ] Linux or Unix
- [ ] Other

### OS versions or Linux distros

Window 11

### Synergy configuration

N/A

### What steps will reproduce the problem?

1. Put Synergy on your second screen
2. Close it
3. Unplug your second screen
4. Open synergy once again
It should not be out of bound and make you unable to access it.

### Log output

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sithlord48**
- fixes #7451

### CODE REVIEW

File: src/gui/src/MainWindow.cpp [hunk 1]
```diff
@@ -110,8 +109,23 @@ void MainWindow::restoreWindow()
 
   const auto &windowPosition = m_AppConfig.mainWindowPosition();
   if (windowPosition.has_value()) {
-    qDebug("restoring main window position");
-    move(windowPosition.value());
+    int x_ = 0;
+    int y_ = 0;
+    int w_ = 0;
+    int h_ = 0;
```
Comment (nbolton) [inline src/gui/src/MainWindow.cpp:115]: What's with the underscores?

**Review by nbolton — COMMENTED** [review]
That's odd. Using GNOME and Wayland, the window isn't restoring to the previous position for me even on `master`. I'll take a deeper look tomorrow.

**Comment by nbolton** [comment]
> That's odd. Using GNOME and Wayland, the window isn't restoring to the previous position for me even on `master`. I'll take a deeper look tomorrow.

On macOS, window position restores fine on master but not on this branch... hmm.

File: src/gui/src/MainWindow.cpp (hunk 1 above)
Comment (sithlord48) [reply src/gui/src/MainWindow.cpp:115]: do you not like them ?

**Comment by sithlord48** [comment]
> That's odd. Using GNOME and Wayland, the window isn't restoring to the previous position for me even on `master`. I'll take a deeper look tomorrow.

This is expected on wayland we can only set our size the app is not really allowed to put windows anywhere it wants

**Comment by sithlord48** [comment]
> > That's odd. Using GNOME and Wayland, the window isn't restoring to the previous position for me even on `master`. I'll take a deeper look tomorrow.
> 
> On macOS, window position restores fine on master but not on this branch... hmm.

That is odd as I had no issues with this branch working on macOS.

**Review by nbolton — APPROVED** [review]

**Pull request merged**